superset/views/core.py

```diff
@@ -744,7 +769,7 @@ def explore_json(self, datasource_type=None, datasource_id=None):
             )
 
             viz_obj = get_viz(
-                datasource_type=datasource_type,
+                datasource_type=cast(str, datasource_type),
```

Comment:
@villebro this is part of the complexity I discussed during the meetup lastweek [here](https://docs.google.com/document/d/1xRLqmUn-G7WiPe8qZnfuvbrYHZ1jmk9aSv9Bhjis2tg/edit#heading=h.wlqxyqcpctfd). Under certain circumstances `datasource_type` can be `None` however in this context it will never to `None` hence the reason for the `cast(...)` which per the [documentation](https://mypy.readthedocs.io/en/stable/casts.html) states, 

> ... give the type checker a little help when it can’t quite understand what is going on.

Reply:
Ok, makes sense.